### ISSUE TITLE
Ferret IOSP scripts drop axes with only one point from the XML description of virtual variables.

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/LAS#975]
**Reported by @noaaroland on 19 Oct 2010 13:59 UTC**
Place this script in a F-TDS installation with a datascan directory for .jnl files (the conf/server/data/dynamic) directory is usually configured in this way.

```
use "http://thredds1.pfeg.noaa.gov:8080/thredds/dodsC/satellite/BA/ssta/mday"
let temp=bassta[y=30:40@ave,x=220:240@ave]
```

Examine the results with dncdump:
```
	float TEMP(TIME) ;
		TEMP:dataset = "1" ;
		TEMP:direction = "L" ;
		TEMP:attribute = "" ;
		TEMP:virtual = "true" ;
		TEMP:ferret_definition = "BASSTA[Y=30:40@AVE,X=220:240@AVE]" ;
	float BASSTA(TIME, ALTITUDE, LAT, LON) ;
		BASSTA:direction = "IJKL" ;
		BASSTA:units = "degree_C" ;
		BASSTA:long_name = "SST, Blended, 0.1 degrees, Global, EXPERIMENTAL" ;
		BASSTA:_FillValue = -9999999.f ;
		BASSTA:missing_value = -9999999.f ;
		BASSTA:ferret_datatype = "FLOAT" ;
		BASSTA:infile_datatype = "FLOAT" ;
		BASSTA:actual_range = "-2.047633 35" ;
		BASSTA:coordsys = "geographic" ;
		BASSTA:fraction_digits = 1.f ;
		BASSTA:numberOfObservations = 3335310.f ;
		BASSTA:percentCoverage = 0.6170671f ;
		BASSTA:standard_name = "sea_surface_temperature" ;
		BASSTA:dataset = "http://thredds1.pfeg.noaa.gov:8080/thredds/dodsC/satellite/BA/ssta/mday" ;
```

The original variable is dimensioned (time, altitude, lat, lon).  The virtual variable is dimensioned (time).  Lat and Lon have been averaged away, but altitude also disappears because it is of length 1.  

The description in the XML looks like this:

```
<data>
<global>
<var name="TEMP" def="BASSTA[Y=30:40@AVE,X=220:240@AVE]">
<attribute name="missing_value" value="-1.00000E+34" />
<grid name="GMI1">
<axes>
<taxis>TIME</taxis>
</axes>
</grid>
</var>
</global>
```

It should look like this:

```
<data>
<global>
<var name="TEMP" def="BASSTA[Y=30:40@AVE,X=220:240@AVE]">
<attribute name="missing_value" value="-1.00000E+34" />
<grid name="GMI1">
<axes>
<zaxis>ALTITUDE</zaxis>
<taxis>TIME</taxis>
</axes>
</grid>
</var>
</global>
```

In fact, if you edit the header.xml to add the z-axis everything works as expected.

Migrated-From: http://dunkel.pmel.noaa.gov/trac/las/ticket/969

**Comment by karlmsmith** [comment]
**Comment by @AnsleyManke on 9 Nov 2010 21:30 UTC**
fixed in Ferret v6.65 (not released but installed on TMAP machines.)

**Comment by karlmsmith** [comment]
**Modified by @kevin-obrien on 9 Nov 2010 22:38 UTC**
